pkg/driver/driver.go

```diff
@@ -111,6 +116,11 @@ func (p *OpenstackDriver) DeleteMachine(ctx context.Context, req *driver.DeleteM
 	klog.V(2).Infof("machine deletion request has been received for %q", req.Machine.Name)
 	defer klog.V(2).Infof("machine deletion request has been processed for %q", req.Machine.Name)
 
+	// Check if incoming CR is a CR we support
+	if req.MachineClass.Provider != openstackProvider {
+		return nil, fmt.Errorf("Requested for Provider '%s', we only support '%s'", req.MachineClass.Provider, openstackProvider)
```

Comment:
Maybe it's better if you return a specific error code like `InvalidArgument` instead of a normal error which likely is categorised as `unknown` error  (cc: @prashanth26) ?

Reply:
100%. I missed to keep this in mind. @AxiomSamarth can you please make this change. And also, would you keep this in mind and make this change as well - https://github.com/gardener/gardener-extension-provider-openstack/pull/262#issuecomment-818732669.